src/test/java/eu/europeana/metis/sandbox/controller/DatasetControllerIT.java

```diff
@@ -84,11 +85,14 @@ public void harvestDatasetWithFile_expectStatus_accepted() {
     ResponseEntity<String> response = makeHarvestingByFile(dataset, null);
     assertEquals(HttpStatus.ACCEPTED, response.getStatusCode());
     assertNotNull(response.getBody());
-    assertTrue(response.getBody().contains("\"dataset-id\":\"1\""));
+    assertTrue(response.getBody().matches("\\{\"dataset-id\":\"\\d\"\\}"));
+    final int expectedDatasetId = extractDatasetId(response.getBody());
+    assertTrue(expectedDatasetId > 0);
 
     // Give time for harvesting to finish to not affect other tests
-    Awaitility.await().atMost(10, MINUTES).until(() -> Objects.requireNonNull(testRestTemplate.getForEntity(getBaseUrl() + "/dataset/{id}",
-            String.class, "1").getBody()).contains("COMPLETED"));
+    Awaitility.await().atMost(10, MINUTES)
```

Comment:
Since the tests will be running in parallel and, I assume, work fine, do we still need to wait for the processing of the dataset to finish?
With the exception of test `retrieveDataset_expectStatus_ok()`

Reply:
Good point, in this one we just want to be sure that the dataset was succesfully created right?

Reply:
done. updated the integration test.